### ISSUE TITLE
[Bug] Vsync and alt tab

### ISSUE THREAD

**Issue by Worzard** [sp614x/optifine#4971]
This issue was mentioned in a comment in #2815 but I presume that it has gone unnoticed. When vsync is enabled the frame cap becomes unlimited and there is no option to set a cap unlike in vanilla. This is a problem because vsync doesn't work when you alt tab causing your gpu to generate unnecessary heat and noise. We should be able to set a frame cap when vsync is enabled so that alt tabbing doesn't cause it to go crazy and degrade performance out of the game.

**Comment by MeowMC** [comment]
This doesn't seem to happen to me. What GPU do you have and what version are you playing?

**Comment by Worzard** [comment]
When I had created this issue I was using the latest version 1.16.3-OptiFine_HD_U_G3. I've updated to 1.16.4-OptiFine_HD_U_G5 and the issue still occurs. It was a bit hard to get this screenshot as the fps drops the moment you alt tab to the game but here it is.
![Screenshot (300)](https://user-images.githubusercontent.com/25337223/98437123-818b1300-20a5-11eb-82db-60a6f6b482de.png)

**Comment by KaiAF** [comment]
Seems to be related to this #5230. Once again, I am not able to reproduce this.

**Comment by Worzard** [comment]
wow, it's been 5 months since someone else replied to this. I have tested this with the latest optifine and minecraft versions and the issue still occurs. I'm not sure how to help with reproducing the issue, but here is a new screenshot of it.
![2021-04-04_08 23 27](https://user-images.githubusercontent.com/25337223/113510424-6ae7cb80-9520-11eb-96de-48f96c339edb.png)

**Comment by MrGcGamer** [comment]
@KaiAF To reproduce it, it is important that minecraft is completely covered by some other window or even on another virtual desktop, it doesn't go into infinity if minecraft is still somewhat on screen.

**Comment by sp614x** [comment]
VSync is implemented in the graphics drivers. OptiFine only enables or disables it, in the same way that vanilla does.
When VSync is enabled the FPS limit is removed as the driver controls the timing and any external delay will disturb it. Vanilla does the same.

**Comment by MrGcGamer** [comment]
How come this only happens with Optifine then?

**Comment by Worzard** [comment]
@sp614x That is how it works in vanilla, but as I stated in my first post , vanilla allows you to set a framerate cap AND have vsync enabled at the same time so that when vsync fails because you are alt tabbed the frame cap kicks in and prevents it from using all the gpu. Here is a screen shot where I highlighted both options for you.
![2021-04-04_19 45 27](https://user-images.githubusercontent.com/25337223/113526192-a87b4180-957e-11eb-884d-95161856263f.png)

**Comment by MrGcGamer** [comment]
As mentioned here: https://github.com/sp614x/optifine/issues/5230#issuecomment-812008174
in Vanilla the frame cap isn't removed when vsync is turned on as opposed to with optifine where it automatically goes to `inf` if vsync is on

**Comment by Lentern** [comment]
> 
> 
> @sp614x That is how it works in vanilla, but as I stated in my first post , vanilla allows you to set a framerate cap AND have vsync enabled at the same time so that when vsync fails because you are alt tabbed the frame cap kicks in and prevents it from using all the gpu. Here is a screen shot where I highlighted both options for you.
> ![2021-04-04_19 45 27](https://user-images.githubusercontent.com/25337223/113526192-a87b4180-957e-11eb-884d-95161856263f.png)

Yeah this issue should be reopened because this isn't present in vanilla.
